src/compiler/moduleSpecifiers.ts

```diff
@@ -376,6 +376,26 @@ namespace ts.moduleSpecifiers {
         if (decl) {
             return decl.name.text;
         }
+
+        // the module could be a namespace, which is default export from an ambient module.
+        /**
+         * declare module "m" {
+         *     namespace ns {
+         *         class c {}
+         *     }
+         *     export = ns;
+         * }
+         */
+        // `import {c} from "m";` is valid, in which case, moduleSymbol is "ns", but the module name should be "m"
+        const exportModuleDeclare = find(moduleSymbol.declarations,
+            d => isModuleDeclaration(d)
+            && d?.parent?.parent?.parent
+            && isModuleBlock(d.parent) && isAmbientModule(d.parent.parent) && isSourceFile(d.parent.parent.parent)
+            && ((d.parent.parent.symbol.exports?.get("export=" as __String)?.valueDeclaration as ExportAssignment).expression as Identifier).escapedText === getTextOfIdentifierOrLiteral(d.name)

```

Comment:
Hm. The text comparison is a little too specific - for example, what if 
```ts
declare module m {
  namespace a.b {
    class c {}
  }
  export = a.b;
}
```
? I think we need to do a symbol identity comparison here. In the checker, this'd ne easy using some `resolveEntityName` or `getSymbolOfNode` machinery - in `moduleSpecifiers.ts` it's maybe a bit more awkward? There's a `getSymbolIfSameReference` function in the checker for seeing if two symbols really refer to the same thing that is what I'd have considered using if it was available. Maybe we need some more context from the checker available? Or just more internals exported from the checker?

Reply:
it is a good sugestion, but I am little worried about something
these functions(`getsymbolofNode`) are in the function `createTypeChecker`, so they could not be exported directly. What about match namespace directly? 
Find the namespace unitil it is not nested namespace -- if module symbol is "b"(it is a nested namespace) and we find its parent "a"(it is namespace and not nested), so the path is "a.b", which matches default export("a.b")

Reply:
> Hm. The text comparison is a little too specific - for example, what if
> 
> ```ts
> declare module m {
>   namespace a.b {
>     class c {}
>   }
>   export = a.b;
> }
> ```
> 
> ? I think we need to do a symbol identity comparison here. In the checker, this'd be easy using some `resolveEntityName` or `getSymbolOfNode` machinery - in `moduleSpecifiers.ts` it's maybe a bit more awkward? There's a `getSymbolIfSameReference` function in the checker for seeing if two symbols really refer to the same thing that is what I'd have considered using if it was available. Maybe we need some more context from the checker available? Or just more internals exported from the checker?

it is a good sugestion, but I am little worried about something
these functions(getsymbolofNode) are in the function createTypeChecker, so they could not be exported directly. What about match namespace directly?
Find the namespace unitil it is not nested namespace -- if module symbol is "b"(it is a nested namespace) and we find its parent "a"(it is namespace and not nested), so the path is "a.b", which matches default export("a.b")

Reply:
Nah, I'm pretty sure you gotta do symbol resolution and matching - otherwise you'll miss renames like
```ts
declare module m {
  namespace a.b {
    class c {}
  }
  import d = a.b;
  export = d;
}
```

Reply:
Thanks a lot for the kind suggestion
Now using getSymbolAtLocation and getAliasedSymbol to get the original symbol of the default export , then compare it with `moduleSymbol`, is this way OK? Did I miss something again?